src/XrdHttp/XrdHttpProtocol.cc

```diff
@@ -1644,6 +1658,13 @@ int XrdHttpProtocol::StartSimpleResp(int code, const char *desc,
     ss << m_staticheaders[""];
   }
 
+  if(xrdcors) {
+    auto corsAllowOrigin = xrdcors->getCORSAllowOriginHeader(CurrentReq.m_origin);
+    if(corsAllowOrigin) {
+      ss << *corsAllowOrigin << crlf;
+    }
```

Comment:
This might be easier to implement if you pass in a reference to `ss` (i.e. the string stream) to the function, which then just adds the origin directly if needed/allowed.

Reply:
It depends, this gives flexibility for the caller of this function. If they don't want to put it in a stringstream but in some other stuff, they can do so?